### ISSUE TITLE
bug: Incorrect ts generation when generating HashMaps that enums as a key.

### ISSUE THREAD

**Issue by MrVintage710** [Aleph-Alpha/ts-rs#349]
**Describe the bug**
While is was working with this library I stumbled in to, as far as I could tell, breaking behavior. I had a setup similar to the following:

```rust
#[derive(TS, Hash, Eq, PartialEq)]
pub enum ComponentId {
  name,
  bulk
}

#[derive(TS)]
pub struct Entity {
  components : HashMap<ComponentId, Component>
}
```

When I translate `ComponentId` I get the expected output:
```typescript
export type ComponentId = "name" | "bulk";
```

However when generating the ts file for `Entity` I get the following:
```typescript
import type { Component } from "./Component";
import type { ComponentId } from "./ComponentId";

export type Entity = { components: { [key: ComponentId]: Component }, };
```
This throws an error as it is not the correct format. Since `ComponentId` is a literal type, the correct translation should be the following:

```typescript
export type Entity = { components: { [key in ComponentId]: Component }, };
```

Notice that after `key`, the colon should be the `in` keyword. Is there away for `ts-rs` to be aware of this in the future? For know I just have a post processing step that makes that change, but it seems like this behavior is not intentional.

**To Reproduce**
1. Go to 'main.rs'
2. Create the following types: 'enum ComponentId { name, bulk }', `enum Component { foo, bar}`, `struct Entity { components : HashMap<ComponentId, Component> }`
3. Derive/implement `TS` for all types made
4. See error in generated ts

**Expected behavior**
When generating the types listed above, `Entity` should generate the following:

```typescript
export type Entity = { components: { [key in ComponentId]: Component }, };
```

**Version**
```toml
ts-rs = "9.0.1"
```

**Comment by NyxCode** [comment]
We fixed this in #339. Can you try the latest master?

**Comment by MrVintage710** [comment]
This worked for me! Sorry I didnt see this change when I was looking around. Thanks for helping me out!
